tools/pubsys-setup/src/main.rs

```diff
@@ -177,15 +176,25 @@ fn run() -> Result<()> {
 /// Searches Infra.toml and expected local paths for a root role and key for the requested repo.
 fn find_root_role_and_key(args: &Args) -> Result<(Option<&PathBuf>, Option<Url>)> {
     let (mut root_role_path, mut key_url) = (None, None);
-
-    if args.infra_config_path.exists() {
-        info!(
-            "Found infra config at path: {}",
-            args.infra_config_path.display()
-        );
-
-        let infra_config =
-            InfraConfig::from_path(&args.infra_config_path).context(error::Config)?;
+    let lock_path = &args
+        .infra_config_path
+        .parent()
+        .context(error::Parent {
+            path: &args.infra_config_path,
+        })?
+        .join("Infra.lock");
+
+    if args.infra_config_path.exists() || lock_path.exists() {
+        let infra_config = if lock_path.is_file() {
+            info!("Found infra lock at path: {}", lock_path.display());
+            InfraConfig::from_lock_path(lock_path).context(error::Config)?
+        } else {
+            info!(
+                "Found infra config at path: {}",
+                args.infra_config_path.display()
+            );
+            InfraConfig::from_path(&args.infra_config_path).context(error::Config)?
+        };
```

Comment:
This looks almost the same as the duplicate code but slightly different. I can't tell if this is actually duplicative.